packages/core/src/codewhisperer/ui/statusBarMenu.ts

```diff
@@ -92,7 +93,9 @@ export function getQuickPickItems(): DataQuickPickItem<string>[] {
         // Add settings and signout
         createSeparator(),
         createSettingsNode(),
-        ...(AuthUtil.instance.isConnected() && !hasVendedIamCredentials() ? [createSignout()] : []),
+        ...(AuthUtil.instance.isConnected() && !(hasVendedIamCredentials() || isSageMaker('SMUS'))
```

Comment:
`hasVendedIamCredentials()` *already* has a check for `isSM` in its own logic. Normally it returns true for SM. Should `hasVendedIamCredentials()` return false for SMUS? If so, change the logic *in hasVendedIamCredentials()`. Not in spread-out places like this.

```typescript
export function hasVendedIamCredentials(isC9?: boolean, isSM?: boolean) {
    isC9 ??= isCloud9()
    isSM ??= isSageMaker('SMAI')
    return isSM || isC9
}
```

Reply:
That is why it should return false, as I mentioned. Then this local special-case wouldn't be needed.

Find a way to avoid this local special-case. What is the *actual semantics* you want to check here?